dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -124,51 +333,100 @@ class MelFilterBankGpu<T>::Impl : public MelFilterImplBase<T> {
+      } else {
+        if (!first)
+          weights_down_norm_[f] = w;
+        if (!last)
+          weights_up_norm_[f] = (1 - w);
+      }
+    }
+    for (int f = fftbin_end_; f < fftbin_size_; f++)
+      fft2mel_[f] = args_.nfilter;
   }
 
   void Setup(ScratchpadEstimator &se, const TensorListShape<> &in_shape) {
-    se.add<mm::memory_kind::device, int>(interval_ends_.size());
-    se.add<mm::memory_kind::device, T>(weights_down_.size());
-    if (args_.normalize)
-      se.add<mm::memory_kind::device, T>(norm_factors_.size());
-
     inner_fft_ = true;
     for (int s = 0; s < in_shape.size(); s++) {
       inner_fft_ &= volume(in_shape.tensor_shape_span(s).begin() + args_.axis + 1,
                            in_shape.tensor_shape_span(s).end()) == 1;
     }
-    fft_dim_ = in_shape.tensor_shape_span(0)[args_.axis];
+    nfft_ = in_shape.tensor_shape_span(0)[args_.axis];
     if (inner_fft_) {
+      se.add<mm::memory_kind::device, int>(fft2mel_.size());
+      se.add<mm::memory_kind::device, T>(weights_down_norm_.size());
+      se.add<mm::memory_kind::device, T>(weights_up_norm_.size());
+
       SetupBlockDescsInnerFft(se, in_shape);
     } else {
+      se.add<mm::memory_kind::device, int>(interval_ends_.size());
+      se.add<mm::memory_kind::device, T>(weights_down_.size());
+      if (args_.normalize)
+        se.add<mm::memory_kind::device, T>(norm_factors_.size());
       SetupBlockDescsOuterFft(se, in_shape);
     }
   }
 
-  void Compute(const T* const* in_list, T **out_list, int ndim,
+  void Compute(T * const*out_list, const T *const *in_list, int ndim,
                Scratchpad *scratchpad, cudaStream_t stream) {
     if (inner_fft_) {
-      FillBlockDescsInnerFft(in_list, out_list);
-    } else {
-      FillBlockDescsOuterFft(in_list, out_list);
-    }
-    BlockDesc<T> *block_descs = nullptr;
-    int *interval_ends = nullptr;
-    T *weights_down = nullptr, *norm_factors = nullptr;
-    std::tie(block_descs, interval_ends, weights_down, norm_factors) =
-          scratchpad->ToContiguousGPU(stream, block_descs_, interval_ends_,
-                                      weights_down_, norm_factors_);
-    if (inner_fft_) {
-      MelFilterBankKernelInnerFft
-          <<<block_descs_.size(), kBlockDim1, 0, stream>>>
-            (block_descs, weights_down, interval_ends, args_.normalize,
-             norm_factors, args_.nfilter, fft_dim_);
+      FillBlockDescsInnerFft(out_list, in_list);
+
+      BlockDesc<T> *block_descs = nullptr;
+      T *weights_down = nullptr, *weights_up = nullptr;
+      int *fft2mel = nullptr;
+      std::tie(block_descs, weights_down, weights_up, fft2mel) =
+            scratchpad->ToContiguousGPU(stream, block_descs_,
+                                        weights_down_norm_, weights_up_norm_,
+                                        fft2mel_);
+
+      int blockHeight = std::min(8, shm_height_);
+      VALUE_SWITCH(shm_height_, ShmHeight, (1, 2, 4, 8, 16, 32), (
```

Comment:
```suggestion
      int blockHeight = std::min(8, shm_height_);
      VALUE_SWITCH(blockHeight, ShmHeight, (1, 2, 4, 8, 16, 32), (
```
you meant to use it here, didn't you? In that case, you can drop 1, 2, 4.

Also a nitpick. use `block_height` as a name

Reply:
No, I meant shm_height_.